jerry-core/api/jerry.c

```diff
@@ -1719,6 +1719,84 @@ jerry_value_to_bigint (const jerry_value_t value) /**< input value */
 #endif /* ENABLED (JERRY_BUILTIN_BIGINT) */
 } /* jerry_value_to_bigint */
 
+/**
+ * Call the integer ecma builtin operation on the api value.
+ *
+ * @return undefined value - if success
+ *         thrown error - otherwise
+ */
+jerry_value_t
+jerry_value_as_integer (jerry_value_t value, /**< input value */
+                        double *number_p)  /**< [out] double number */
+{
+  jerry_assert_api_available ();
+
+  if (ecma_is_value_error_reference (value))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  return jerry_return (ecma_op_to_integer (value, number_p));
```

Comment:
`ecma_op_to_integer` returns `ECMA_VALUE_EMPTY` on success which is not compatible with `jerry_return`. Handle the error case separately and return with undefined otherwise.